src/host/getset.cpp

```diff
@@ -1347,6 +1347,33 @@ void DoSrvPrivateAllowCursorBlinking(SCREEN_INFORMATION& screenInfo, const bool
     return Status;
 }
 
+// Routine Description:
+// - A private API call for performing a line feed, possibly preceded by carriage return.
+//    Moves the cursor down one line, and possibly also to the leftmost column.
+// Parameters:
+// - screenInfo - A pointer to the screen buffer that should perform the line feed.
+// - withReturn - Set to true if a carriage return should be performed as well.
+// Return value:
+// - STATUS_SUCCESS if handled successfully. Otherwise, an appropriate status code indicating the error.
+[[nodiscard]] NTSTATUS DoSrvPrivateLineFeed(SCREEN_INFORMATION& screenInfo, const bool withReturn)
+{
+    auto& textBuffer = screenInfo.GetTextBuffer();
+    auto cursorPosition = textBuffer.GetCursor().GetPosition();
+
+    textBuffer.GetCursor().SetIsOn(true);

```

Comment:
nit: this might be worth explicitly calling out with a comment; I'd hate it to get lost in a later refactor

Reply:
I'm not actually sure why it's necessary though, or even if it's really necessary. I was just copying the behavior from the previous `WriteCharsLegacy` codepath. But I'll do some more research and see if I can come up with something meaningful to say about it.

Reply:
OK, it seems it's necessary to turn the cursor on to prevent old copies from sometimes being left behind when the screen scrolls. I've now added a comment explaining that.